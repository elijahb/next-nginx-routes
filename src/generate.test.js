"use strict";

const { readFileSync } = require("fs");
const generate = require('./generate')

test("generate routes for example project", () => {
  const nginxRoutes = generate(readFileSync(__dirname + "/routes-manifest.example.json", "utf8"))
  expect(nginxRoutes).toMatchSnapshot();
});
