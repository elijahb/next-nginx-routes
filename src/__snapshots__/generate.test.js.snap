// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate routes for example project 1`] = `
"
location ~ ^/(?:/)?$ {
    try_files /index.html =404;
}

location ~ ^/hello/world(?:/)?$ {
    try_files /hello/world.html =404;
}

location ~ ^/([^/]+?)/([^/]+?)/([^/]+?)(?:/)?$ {
    try_files /[foo]/[bar]/[id].html =404;
}

location ~ ^/([^/]+?)/([^/]+?)/foo(?:/)?$ {
    try_files /[foo]/[bar]/foo.html =404;
}

location ~ ^/([^/]+?)(?:/)?$ {
    try_files /[foo].html =404;
}"
`;
